plugins/pulp_docker/plugins/registry.py

```diff
@@ -368,8 +368,39 @@ def get_manifest(self, reference):
         :return:          A 2-tuple of the digest and the manifest, both basestrings
         :rtype:           tuple
         """
+        manifests = []
+        request_headers = {}
+        content_type_header = 'content-type'
+        schema1 = 'application/vnd.docker.distribution.manifest.v1+json'
+        schema2 = 'application/vnd.docker.distribution.manifest.v2+json'
         path = self.MANIFEST_PATH.format(name=self.name, reference=reference)
-        headers, manifest = self._get_path(path)
+        # set the headers for first request
+        request_headers['Accept'] = schema2
+        headers, manifest = self._get_path(path, headers=request_headers)
```

Comment:
ok i think i got confused about which headers we are talking about, if it is a parameter from _get_path() then those are request headers, and the ones which _get_path returns are response_headers. I think now we are on the same page and i completely agree with your suggestion.